File: src/net/dinkla/raytracer/utilities/Resolution.scala
package net.dinkla.raytracer.utilities

/**
 * Created by Dinkla on 09.07.2015.
 */
class Resolution(val hres: Int, val vres: Int) {

  def this(vres: Int) = this((vres / 9.0 * 16).toInt, vres)

  override def toString: String = s"($hres,$vres)"

}

object Resolution {

  val RESOLUTION_32: Resolution = new Resolution(32)
  val RESOLUTION_320: Resolution = new Resolution(320)
  val RESOLUTION_480: Resolution = new Resolution(480)
  val RESOLUTION_720: Resolution = new Resolution(720)
  val RESOLUTION_1080: Resolution = new Resolution(1080)
  val RESOLUTION_1440: Resolution = new Resolution(1440)
  val RESOLUTION_2160: Resolution = new Resolution(2160)

}

File: src/net/dinkla/raytracer/utilities/Resolution.java
package net.dinkla.raytracer.utilities;

/**
 * Created by IntelliJ IDEA.
 * User: jorndinkla
 * Date: 03.05.2010
 * Time: 09:30:40
 */
public class Resolution {

    public static Resolution RESOLUTION_32 = new Resolution(32);
    public static Resolution RESOLUTION_320 = new Resolution(320);
    public static Resolution RESOLUTION_480 = new Resolution(480);
    public static Resolution RESOLUTION_720 = new Resolution(720);
    public static Resolution RESOLUTION_1080 = new Resolution(1080);
    public static Resolution RESOLUTION_1440 = new Resolution(1440);
    public static Resolution RESOLUTION_2160 = new Resolution(2160);

    // Resolution
    public final int hres;
    public final int vres;

    public Resolution(final int vres) {
        this.hres = (int) (vres / 9 * 16);
        this.vres = vres;
    }

    public Resolution(final int hres, final int vres) {
        this.hres = hres;
        this.vres = vres;
    }

    @Override
    public String toString() {
        return "(" + hres + "," + vres + ")";
    }
}
